tools/build_api.py

```diff
@@ -674,6 +674,12 @@ def build_mbed_libs(target, toolchain_name, options=None, verbose=False, clean=F
         toolchain.jobs = jobs
         toolchain.build_all = clean
 
+        # Take into account the library configuration (MBED_CONFIG_FILE)
+        config = Config(target)
+        toolchain.config = config
+        config.add_config_files([MBED_CONFIG_FILE])
+        toolchain.set_config_data(toolchain.config.get_config_data())
```

Comment:
That feels a bit on the dark magic side :). Initially, `config` was not supposed to be an attribute of `toolchain` at all. Commit ac86584c added it as a convenience, so that you don't have to pass around `config` anymore, just `toolchain`. I'd keep it just as a convenience and make the calls to `set_config_data` explicit, so that it's clear what happens in the code.